src/compositions/cascade/useCascadePanel.ts

```diff
@@ -0,0 +1,53 @@
+import { computed, inject, InjectionKey, MaybeRefOrGetter, provide, Ref, toRef } from 'vue'
+import { CascadePanelId, getInjectedCascadePanels } from '@/compositions'
+
+export const cascadePanelKey: InjectionKey<UseCascadePanel> = Symbol('UseCascadePanel')
+
+export type UseCascadePanel = {
+  isOpen: Ref<boolean>,
+  close: () => void,
+  open: () => void,
+  toggle: () => void,
+}
+
+function getInjectedPanel(): UseCascadePanel {
+  const cascadePanel = inject(cascadePanelKey)
+
+  if (!cascadePanel) {
+    throw new Error('Cascade panel not found. Are you sure the component calling useCascadePanel() exists within a <p-cascade-panel> component context?')
+  }
+
+  return cascadePanel
+}
+
+export function useCascadePanel(id?: MaybeRefOrGetter<CascadePanelId>): UseCascadePanel {
+  if (!id) {
+    return getInjectedPanel()
+  }
```

Comment:
What's the use case for using this with or without an id?

Reply:
This is another devex piece - so the `PCascadePanel` component wrapper can call `useCascadePanel(panelId)`, which will `provide` the hydrated panel without needing to passing around ids. Then in a downstream component you can do `useCascadePanel` without any further injection/knowledge of the id